generated/niscope/niscope/session.py

```diff
@@ -1997,6 +1996,164 @@ def fetch(self, num_samples=None, relative_to=enums.FetchRelativeTo.PRETRIGGER,
 
         return wfm_info
 
+    @ivi_synchronized
+    def fetch_array_measurement(self, array_meas_function, timeout=hightime.timedelta(seconds=5.0), other_channel=None):
+        r'''fetch_array_measurement
+
+        Obtains a waveform from the digitizer and returns the specified
+        measurement array. This method may return multiple waveforms depending
+        on the number of channels, the acquisition type, and the number of
+        records you specify.
+
+        Note:
+        Some functionality, such as time stamping, is not supported in all
+        digitizers.
+
+        Tip:
+        This method requires repeated capabilities. If called directly on the
+        niscope.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        niscope.Session repeated capabilities container, and calling this method on the result.
+
+        Args:
+            array_meas_function (enums.ArrayMeasurement): The array measurement to perform.
+
+            timeout (hightime.timedelta, datetime.timedelta, or float in seconds): The time to wait in seconds for data to be acquired; using 0 for this
+                parameter tells NI-SCOPE to fetch whatever is currently available. Using
+                -1 for this parameter implies infinite timeout.
+
+            other_channel (str): The identifier for the "other channel" for multi-channel measurements such as Add Channels or Multiply Channels.
```

Comment:
I think we're better off not adding this parameter:
- Breaks the design of keeping all channel-based parameters as repeated capabilities
- There are other `array_meas_function`s for which a property is expected to be set before calling this method. e.g. `POLYNOMIAL_INTERPOLATION` requires `meas_polynomial_interpolation_order` property to be set. It's inconsistent to include `other_channel` in the method but not those (unless I'm missing something)

Also, if we do decide to keep the `other_channel` property, then
- We should do the same for `add_waveform_processing` method too
- Consider making `meas_other_channel` property private

@marcoskirsch, what do you think?

Reply:
On my part, I agree that it should probably be removed as a parameter. I think I added it because I ran into problems with needing to set other channels when first experimenting with the measurement library and thought that would be a "clever" change, but in terms of consistency not including it is better.

Reply:
> I think we're better off not adding this parameter 

So this was a clever way of preventing customer from having to set the property?

```
        with _NoChannel(session=self):
              if other_channel is not None:
                  self.meas_other_channel = other_channel
```

I agree this is a bad idea. It's a "side effect": a user would be surprised that calling a function changed the value of a property behind his back. Unexpected things like this are best avoided.

If we are convinced that we're better off keeping the parameter, it would only be ok to do if we make the property private. I don't think we're there. Safer to map the C API closer.

Last thing - anywhere the user sets a channel, they should be able to set it as an int, etc. We allow this for the repeated capability and we should make it work for any property that expects a channel string in the C runtime.